api/capi/src/nnstreamer-capi-single.c

```diff
@@ -597,7 +636,10 @@ ml_single_set_timeout (ml_single_h single, unsigned int timeout)
 
   single_h = (ml_single *) single;
 
+  g_mutex_lock (&single_h->lock);
```

Comment:
Please, note that the mutex cost is not inexpensive even though the modern Linux has introduced the FUTEX mechanism.